src/workflow.py

```diff
@@ -1,80 +1,42 @@
 # -*- coding: utf-8 -*-
 
-import os
-import configparser
 import time
 
 import alfred
 import otp
+from storage.plain_text_storage import PlainTextStorage
+from storage.apple_keychain_storage import AppleKeychainStorage
+from storage.exceptions import StorageException
 
 
 class AlfredGAuth(alfred.AlfredWorkflow):
-    _config_file_initial_content = """
-#Examples of valid configurations:
-#[google - bob@gmail.com]
-#secret=xxxxxxxxxxxxxxxxxx
-#
-#[evernote - robert]
-#secret=yyyyyyyyyyyyyyyyyy
-"""
-
     _reserved_words = ['add', 'update', 'remove']
 
-    def __init__(self, config_file='~/.gauth', max_results=20):
+    def __init__(self, max_results=20):
         self.max_results = max_results
 
-        self._config_file = config_file
-        self.config_file = os.path.expanduser(self._config_file)
-        self.config = configparser.RawConfigParser()
-        self.config.read(self.config_file)
+        self._init_storage()
 
-        # If the configuration file doesn't exist, create an empty one
-        if not os.path.isfile(self.config_file):
-            self.create_config()
+    def _init_storage(self):
+        self._storage = AppleKeychainStorage()
 
         try:
-            if not self.config.sections():
-                # If the configuration file is empty,
-                # tell the user to add secrets to it
-                self.write_item(self.config_file_is_empty_item())
-                return
-        except Exception as e:
+            self._storage.validate_storage()
+        except StorageException as e:
             item = self.exception_item(title='{}: Invalid syntax'
-                                       .format(self._config_file),
-                                       exception=e)
+                                       .format(str(e)),
+                                       exception=e.__context__)
             self.write_item(item)
 
-    def create_config(self):
-        with open(self.config_file, 'w') as f:
-            f.write(self._config_file_initial_content)
-            f.close()
-
     def config_get_account_token(self, account):
         try:
-            secret = self.config.get(account, 'secret')
-        except:
-            secret = None
-
-        try:
-            key = self.config.get(account, 'key')
-        except:
-            key = None
-
-        try:
-            hexkey = self.config.get(account, 'hexkey')
-        except:
-            hexkey = None
```

Comment:
Removed unused code for dealing with `key` and `hexkey` configuration variables.